qt/VariantHelpers.cc

```diff
@@ -178,8 +178,11 @@ bool change(TrackerStat& setme, tr_variant const* value)
         changed = true;
     }
 
-    if (site_changed && !setme.sitename.isEmpty() && !setme.announce.isEmpty() && trApp != nullptr)
+    if (site_changed && !setme.announce.isEmpty() && trApp != nullptr)
     {
+        QStringList const separated_host = QUrl{ setme.announce }.host().split(QStringLiteral("."));
+
+        setme.sitename = separated_host.at(separated_host.size() - 2);
```

Comment:
What if this was wrapped in a `if (setme.sitename.isEmpty())`?

That way we could avoid unnecessary work when a 4.x client is talking to a 4.x server

Reply:
In the edge case this is trying to fix the sitename will always be empty because 3.xx servers don't know what a sitename is. On possible solution is to check the server version instead.

Reply:
> In the edge case this is trying to fix the sitename will always be empty because 3.xx servers don't know what a sitename is. 

Yes, another (more common?) case is that it's _not_ empty because client and server are both 4.x, and this PR creates extra unnecessary work for that case.

IIUC it's empty when there's a 3.x server but nonempty when it's a 4.x server, correct? If so, then I still like the idea of populating it only if it's empty. `isEmpty()` is practically free; much cheaper than QUrl instantiation + splitting to a stringlist :smile_cat:

Reply:
Sorry, I didn't interpret your suggestion correctly at first. Should be fixed now.